Dockerfile.rhel7

```diff
@@ -1,25 +1,26 @@
-# THIS FILE IS GENERATED FROM Dockerfile DO NOT EDIT
-# TODO switch the default image to rhel9 and drop the rhel8 one in 4.15 because
-# we can require by the time we get to 4.14 that we don't have any rhel8 hosts left
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.21-openshift-4.16 AS builder
+# Use RHEL 9 as the primary builder base for the Machine Config Operator
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16 AS builder
 ARG TAGS=""
 WORKDIR /go/src/github.com/openshift/machine-config-operator
 COPY . .
 # FIXME once we can depend on a new enough host that supports globs for COPY,
 # just use that.  For now we work around this by copying a tarball.
 RUN make install DESTDIR=./instroot && tar -C instroot -cf instroot.tar .
 
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16 AS rhel9-builder
+# Add a RHEL 8 builder to compile the RHEL 8 compatible binaries
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.21-openshift-4.16 AS rhel8-builder
 ARG TAGS=""
 WORKDIR /go/src/github.com/openshift/machine-config-operator
 COPY . .
-RUN make install DESTDIR=./instroot
+RUN make install DESTDIR=./instroot-rhel8 && tar -C instroot-rhel8 -cf instroot-rhel8.tar .
 
-FROM registry.ci.openshift.org/ocp/4.16:base
+# Use the RHEL 9 base for the final image and copy over both RHEL 9 and RHEL 8 binaries
```

Comment:
We are only copying `machine-config-daemon` binary as `machine-config-daemon.rhel8` binary  which is RHEL 8 based, rest binaries are RHEL 9 based . We can update comment accordingly.